PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseOfBlockOnAsyncClientsCheckTest.java

```diff
@@ -0,0 +1,175 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+import com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool.UseOfBlockOnAsyncClientsCheck.UseOfBlockOnAsyncClientsVisitor;
+
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class is used to test the UseOfBlockOnAsyncClientsCheck class.
+ * The UseOfBlockOnAsyncClientsCheck class is an inspection tool that checks for the use of block() method on async clients in Azure SDK.
+ * This inspection will check for the use of block() method on reactive types like Mono, Flux, etc.
+ * 1. Flux<ByteBuffer> downloadFlux = blobAsyncClient.download()
+ *                 .doOnComplete(() -> System.out.println("Download complete"))
+ *                 .doOnError(error -> System.err.println("Error downloading file: " + error.getMessage()));
+ *
+ *             // This call should be flagged
+ *             downloadFlux.blockFirst();
+ *
+ * 2. // Download a file from Azure Blob Storage using the async client
+ *             Mono<Void> downloadMono = blobAsyncClient.downloadToFile("downloadedFilePath")
+ *                 .doOnSuccess(response -> System.out.println("File downloaded successfully"))
+ *                 .doOnError(error -> System.err.println("Error downloading file: " + error.getMessage()));
+ *
+ *             // This call should also be flagged by the inspection tool
+ *             downloadMono.blockOptional();
+ */
+public class UseOfBlockOnAsyncClientsCheckTest {
+
+    // Declare as instance variables
+    @Mock
+    private ProblemsHolder mockHolder;
+
+    @Mock
+    private JavaElementVisitor mockVisitor;
+
+    @Mock
+    private PsiMethodCallExpression mockElement;
+
+    @BeforeEach
+    public void setup() {
+        mockHolder = mock(ProblemsHolder.class);
+        mockVisitor = createVisitor();
+        mockElement = mock(PsiMethodCallExpression.class);
+    }
+
+    /**
+     * This is the main test method that tests the KustoQueriesWithTimeIntervalInQueryStringCheck class.
+     */
+    @Test
+    public void testUseOfBlockOnAsyncClientsVisitor() {
+
+        // assert visitor
+        assertVisitor();
```

Comment:
Do we need this?

Reply:
eh not really its just an extra step to check the visitor is initialised correctly

Reply:
This is not required as this is testing the test code instead of the source code.

Reply:
will take out